src/Doctrine/Orm/Filter/AttributeFilter.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusGraphqlPlugin\Doctrine\Orm\Filter;
+
+use ApiPlatform\Core\Api\FilterInterface;
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\AbstractContextAwareFilter;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\PropertyHelperTrait;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use Sylius\Component\Attribute\AttributeType\CheckboxAttributeType;
+use Sylius\Component\Attribute\AttributeType\DateAttributeType;
+use Sylius\Component\Attribute\AttributeType\DatetimeAttributeType;
+use Sylius\Component\Attribute\AttributeType\IntegerAttributeType;
+use Sylius\Component\Attribute\AttributeType\PercentAttributeType;
+use Sylius\Component\Attribute\AttributeType\SelectAttributeType;
+use Sylius\Component\Attribute\Model\AttributeInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * Filters the collection by value of given attribute IRI
+ */
+class AttributeFilter extends AbstractContextAwareFilter implements FilterInterface
+{
+    use PropertyHelperTrait;
+
+    private IriConverterInterface $iriConverter;
+
+    public const OPERATOR_EXACT = 'exact';
+
+    public const OPERATOR_PARTIAL = 'partial';
+
+    public const ATTRIBUTE_ID = 'attribute_id';
+
+    public const VALUE = 'value';
+
+    public function __construct(
+        ManagerRegistry $managerRegistry,
+        IriConverterInterface $iriConverter,
+        ?RequestStack $requestStack = null,
+        LoggerInterface $logger = null,
+        array $properties = null,
+        NameConverterInterface $nameConverter = null
+    ) {
+        parent::__construct(
+            $managerRegistry,
+            $requestStack,
+            $logger,
+            $properties,
+            $nameConverter
+        );
+
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * @param mixed $value
+     */
+    protected function filterProperty(
+        string $property,
+        $value,
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        string $operationName = null
+    ): void {
+        if (
+            !\is_array($value) ||
+            !$this->isPropertyEnabled($property, $resourceClass)
```

Comment:
Maybe a more descriptive private function?

Reply:
Check extracted to private method